### PR TITLE
Resize home page images

### LINKED ISSUE #3994: Optimize home page media assets
## Expected behavior
The site doesn't weigh ~10MB on the home page.  
## Current behavior
Some of the home-* images are big, ranging from 781KB to 564KB. Also the featured project image is 239KB. These can all be resized / optimized for the view port.

## Steps to replicate
Open dev tools, turn off caching and look at the network requests for Img assets. 
The reason i was surprised was the cost / weight of data on our logged out home page -
 https://whatdoesmysitecost.com/test/170822_HC_6e97d744f72cdfc2b6439a51c8418afc#usdCost

We should check all our assets are optimized (both size / format / quality) to reduce any network overhead on our site. This should apply for any new PR's that add media assets to the site.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Fixes #3994 .

Describe your changes.
Renders home page images using the Thumbnail component, with a reasonable maximum width set.

Ideally, we should not be committing large images into this repo in the first place, so the originals might need some optimisation.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://resize-images.pfe-preview.zooniverse.org/
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

## Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Thumbnail component should be updated so that it doesn't require the full URL for an image.

File: app/pages/home-common/featured-project.jsx [hunk 1]
```diff
@@ -10,14 +10,16 @@ Featured Project Component
...
 
 const FeaturedProject = (() =>
   <section className="home-featured">
     <h1 className="secondary-kicker">Featured Project</h1>
     <div className="home-featured-images">
-      <img
+      <Thumbnail
         alt="Steller Sea Lion"
-        src="/assets/featured-projects/featured-project-20170801-steller.jpg"
+        src="//www.zooniverse.org/assets/featured-projects/featured-project-20170801-steller.jpg"
```
Comment (camallen) [inline app/pages/home-common/featured-project.jsx]: This won't work for preview branch deploys if the assets aren't in the www.zooniverse.org bucket path due to the nginx rewrites.

Comment (eatyourgreens) [reply app/pages/home-common/featured-project.jsx]: Turns out mapping staging branches to S3 bucket paths isn't simple. I'm not sure if that's something that should be handled with some extra logic in the `Thumbnail` component, or deferred to the backend somewhere.

If it's handled in the javascript, we'll need a new component method that can take a path from the image src, determine whether or not it contains a hostname, map that hostname to an S3 bucket path then return the S3 path.

**Comment by eatyourgreens** [comment]
Rebased to updated the featured project.

**Comment by simoneduca** [comment]
Is this ready to go or Cam's comment about deploy preview branches is a blocker?

**Comment by eatyourgreens** [comment]
Git hub hid my comment about staging branches (from last week), because it was made on an outdated commit, so here it is again.

> Turns out mapping staging branches to S3 bucket paths isn't simple. I'm not sure if that's something that should be handled with some extra logic in the Thumbnail component, or deferred to the backend somewhere.

> If it's handled in the javascript, we'll need a new component method that can take a path from the image src, determine whether or not it contains a hostname, map that hostname to an S3 bucket path then return the S3 path.

**Comment by eatyourgreens** [comment]
Rebased (and staged again) to update the featured project.

File: app/pages/home-not-logged-in/discover.jsx [hunk 2]
```diff
@@ -53,7 +54,7 @@ const HomePageDiscover = (({ showDialog }) =>
     </div>
 
     <div className="home-discover__image">
-      <img role="presentation" src="/assets/home-computer.png" />
+      <Thumbnail role="presentation" src="//www.zooniverse.org/assets/home-computer.png" width={450} />
```
Comment (simoneduca) [inline app/pages/home-not-logged-in/discover.jsx]: Why did you switch to an absolute url?

Comment (eatyourgreens) [reply app/pages/home-not-logged-in/discover.jsx]: You can't pass a relative URL to the Thumbnail component.

Comment (eatyourgreens) [reply app/pages/home-not-logged-in/discover.jsx]: The generated image URL ends up being something like https://thumbnails.zooniverse.org/450x999/www.zooniverse.org/assets/home-computer.png

Although I'm not seeing that for the featured project so I guess thumbnails.zooniverse.org expects the first part to be an actual bucket name on S3.

Comment (simoneduca) [reply app/pages/home-not-logged-in/discover.jsx]: I don't think it's ideal in general to use absolute urls. But /shrug.

Comment (eatyourgreens) [reply app/pages/home-not-logged-in/discover.jsx]: That was Cam's comment too.

It turns out to  be a bucket path on S3. I'm interested in hearing ideas on how to fix that for staging branches. See my comment above.

**Pull request merged**